pyomo/contrib/parmest/parmest.py

```diff
@@ -848,99 +910,117 @@ def leaveNout_bootstrap_test(self, lNo, lNo_samples, bootstrap_samples,
-                                    distribution=distribution, alphas=alphas, 
+
+            training, test = self.confidence_region_test(bootstrap_theta,
+                                    distribution=distribution, alphas=alphas,
                                     test_theta_values=theta)
-                
+
             results.append((sample, test, training))
-        
+
         # Reset callback_data (back to full data set)
         self.callback_data = data
-        
+
         return results
-    
-    
-    def objective_at_theta(self, theta_values):
+
+
+    def objective_at_theta(self, theta_values=None,
+                            initialize_parmest_model = False):
         """
         Objective value for each theta
 
         Parameters
         ----------
         theta_values: pd.DataFrame, columns=theta_names
             Values of theta used to compute the objective
-            
+
+        initialize_parmest_model: boolean
+            True: Solve square problem instance, build extensive form of the model for 
+            parameter estimation, and set flag model_initialized to True
+
+
         Returns
         -------
         obj_at_theta: pd.DataFrame
-            Objective value for each theta (infeasible solutions are 
+            Objective value for each theta (infeasible solutions are
             omitted).
         """
-        assert isinstance(theta_values, pd.DataFrame)
-        
-        # for parallel code we need to use lists and dicts in the loop
-        theta_names = theta_values.columns
-        all_thetas = theta_values.to_dict('records')
+        if theta_values is None:
+            all_thetas = {} # dictionary to store fitted variables
+            theta_names = self.theta_names
+        else:
+            assert isinstance(theta_values, pd.DataFrame)
+
+            # for parallel code we need to use lists and dicts in the loop
+            theta_names = theta_values.columns
+            all_thetas = theta_values.to_dict('records')
+
         task_mgr = utils.ParallelTaskManager(len(all_thetas))
-        local_thetas = task_mgr.global_to_local_data(all_thetas)
-        
+        if all_thetas:
+            local_thetas = task_mgr.global_to_local_data(all_thetas)
+
         # walk over the mesh, return objective function
         all_obj = list()
-        for Theta in local_thetas:
-            obj, thetvals, worststatus = self._Q_at_theta(Theta)
+        if len(all_thetas) > 0:
+            for Theta in local_thetas:
+                obj, thetvals, worststatus = self._Q_at_theta(Theta, initialize_parmest_model=initialize_parmest_model)
+                if worststatus != pyo.TerminationCondition.infeasible:
+                     all_obj.append(list(Theta.values()) + [obj])
+                # DLW, Aug2018: should we also store the worst solver status?
+        else:
+            obj, thetvals, worststatus = self._Q_at_theta(thetavals={}, initialize_parmest_model=initialize_parmest_model)
```

Comment:
When `theta_values` is None, the fixed values are used, and the returned `obj_at_theta` is an empty dataframe.  Can `obj_at_theta` return the fixed theta values and objective value even when `theta_values` is None?